packages/server-wallet/src/wallet/store.ts

```diff
@@ -1,14 +1,29 @@
 import Objection from 'objection';
+import {ChannelResult} from '@statechannels/client-api-schema';
 
 import {SignState} from '../protocols/actions';
 import {Channel, SyncState} from '../models/channel';
+import {Bytes32} from '../type-aliases';
+import {ChannelState} from '../protocols/state';
 
 export const Store = {
-  signState: async function(action: SignState, tx: Objection.Transaction): Promise<SyncState> {
+  signState: async function(
+    action: SignState,
+    tx: Objection.Transaction
+  ): Promise<{outgoing: SyncState; channelResult: ChannelResult}> {
     const channel = await Channel.forId(action.channelId, tx);
-    const notifications = channel.signAndAdd(action);
+    const outgoing = channel.signAndAdd(action);
     await Channel.query(tx).update(channel);
 
-    return notifications;
+    const {channelResult} = channel;
+
+    return {outgoing, channelResult};
+  },
+
+  getChannel: async function(
+    channelId: Bytes32,
+    tx: Objection.Transaction | undefined
+  ): Promise<ChannelState | undefined> {
+    return (await Channel.forId(channelId, tx))?.protocolState;
```

Comment:
Wouldn't this throw an exception if `(await Channel.forId(channelId, tx))` was `undefined`?

Reply:
No, it returns `Promise.resolve(undefined)`!
```
❯ ts-node
> const foo: {bar: undefined | {baz: 'baz'}} = {bar: undefined}
'use strict'
> foo.bar?.baz
undefined
```

Reply:
Ah, it's a TypeScript 3.7 [feature](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining)